### PR TITLE
initial nix configuration for byron-proxy repo

### PR DESCRIPTION
**Pull request by disassembler**

### CODE REVIEW

File: .stack-to-nix.cache [hunk 1]
```diff
@@ -0,0 +1,49 @@
+https://github.com/input-output-hk/iohk-monitoring-framework 3ac2191fc4c68b46b7f6c08e2873f71949cc68e7 iohk-monitoring 1yajic3x46jichj57vz3cqyvn5cdnayc4pccmcmds902wg7nqgia iohk-monitoring iohk-monitoring.nix
```
Comment (erikd) [inline .stack-to-nix.cache]: Is this an autogenerated file? If so, does it **need** to be in the repo?

**Review by erikd — APPROVED** [review]
Seems to be a **lot** of autogenerated files. If any of these could be removed it would be a goof idea to do so.

If they can't be removed then a reluctant :+1:

File: nix/regenerate.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+
+exec $(nix-build `dirname $0`/../lib.nix -A nix-tools.regeneratePackages --no-out-link --option substituters "https://hydra.iohk.io https://cache.nixos.org" --option trusted-substituters "" --option trusted-public-keys "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=")
```
Comment (cleverca22) [inline nix/regenerate.sh]: setting `substituters` when your not a trusted nix user will cause the build to fail due to lackof permissions

**Review by cleverca22 — APPROVED** [review]

**Comment by disassembler** [comment]
bors ping

**Pull request merged**